File: Bhatkanti/Models/Booking_Payment_Details.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Bhatkanti.Models
{
    [Table("booking_payment_details")]
    public class Booking_Payment_Details
    {
        [Key]
        [StringLength(14)]
        public string Payment_ID { get; set; }

        [ForeignKey("Booking")]
        public int? Booking_ID { get; set; }

        [Required]
        public DateTime Transaction_Date { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;

        [StringLength(100)]
        public string Bank_Transaction_Code { get; set; }

        [Required]
        [StringLength(50)]
        public TransactionMode Transaction_Mode { get; set; } // UPI/NetBanking/Cards

        [StringLength(20)]
        public PaymentStatus? Status { get; set; } = null;

        [StringLength(255)]
        public string Remarks { get; set; }

        // Navigation Properties
        public User_Guide_Bookings Booking { get; set; }

        public Booking_Payment_Details()
        {
            // Payment_ID will be set by the service or controller
            //public class PaymentService
            //{
            //    private readonly ApplicationDbContext _context;

            //    public PaymentService(ApplicationDbContext context)
            //    {
            //        _context = context;
            //    }

            //    public string GeneratePaymentID()
            //    {
            //        string datePart = DateTime.UtcNow.ToString("yyyyMMdd"); // 20240805
            //        int sequenceNumber = _context.BookingPaymentDetails
            //            .Count(bp => bp.Payment_ID.StartsWith(datePart)) + 1;

            //        string sequence = sequenceNumber.ToString("D6"); // 000001 format

            //        return datePart + sequence;
            //    }
            //}
        }

        public enum TransactionMode
        {
            UPI,
            NetBanking,
            Cards
        }

        public enum PaymentStatus
        {
            Success,
            Failure
        }
    }
}

File: Backend SpringBoot/src/main/java/com/explorer/Models/BookingPaymentDetails.java
package com.explorer.Models;

import jakarta.persistence.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "bookingpaymentdetails")
public class BookingPaymentDetails {

	@Id
	@Column(length = 14)
	private String paymentId;

	@ManyToOne
	@JoinColumn(name = "booking_id")
	private UserGuideBookings booking;

	private LocalDateTime transactionDate;

	@Column(columnDefinition = "TIMESTAMP DEFAULT CURRENT_TIMESTAMP")
	private LocalDateTime timestamp;

	@Column(length = 100)
	private String bankTransactionCode;

	@Enumerated(EnumType.STRING)
	@Column(length = 50)
	private TransactionMode transactionMode;

	@Enumerated(EnumType.STRING)
	@Column(length = 20)
	private PaymentStatus status;

	@Column(length = 255)
	private String remarks;

	@Column(precision = 10, scale = 2) // Assuming up to 10 digits, with 2 decimal places
	private BigDecimal amount;

	// Enum types for TransactionMode and PaymentStatus
	public enum TransactionMode {
		UPI, NETBANKING, CARDS
	}

	public enum PaymentStatus {
		SUCCESS, FAILURE
	}

	public BookingPaymentDetails(String paymentId, UserGuideBookings booking, LocalDateTime transactionDate,
			LocalDateTime timestamp, String bankTransactionCode, TransactionMode transactionMode, PaymentStatus status,
			String remarks, BigDecimal amount) {
		super();
		this.paymentId = paymentId;
		this.booking = booking;
		this.transactionDate = transactionDate;
		this.timestamp = timestamp;
		this.bankTransactionCode = bankTransactionCode;
		this.transactionMode = transactionMode;
		this.status = status;
		this.remarks = remarks;
		this.amount = amount;
	}

	public BookingPaymentDetails() {

	}

	public String getPaymentId() {
		return paymentId;
	}

	public void setPaymentId(String paymentId) {
		this.paymentId = paymentId;
	}

	public UserGuideBookings getBooking() {
		return booking;
	}

	public void setBooking(UserGuideBookings booking) {
		this.booking = booking;
	}

	public LocalDateTime getTransactionDate() {
		return transactionDate;
	}

	public void setTransactionDate(LocalDateTime transactionDate) {
		this.transactionDate = transactionDate;
	}

	public LocalDateTime getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(LocalDateTime timestamp) {
		this.timestamp = timestamp;
	}

	public String getBankTransactionCode() {
		return bankTransactionCode;
	}

	public void setBankTransactionCode(String bankTransactionCode) {
		this.bankTransactionCode = bankTransactionCode;
	}

	public TransactionMode getTransactionMode() {
		return transactionMode;
	}

	public void setTransactionMode(TransactionMode transactionMode) {
		this.transactionMode = transactionMode;
	}

	public PaymentStatus getStatus() {
		return status;
	}

	public void setStatus(PaymentStatus status) {
		this.status = status;
	}

		
	public String getRemarks() {
		return remarks;
	}

	public void setRemarks(String remarks) {
		this.remarks = remarks;
	}

	public BigDecimal getAmount() {
		return amount;
	}

	public void setAmount(BigDecimal amount) {
		this.amount = amount;
	}

}
